### PR TITLE
fix(ludicrous): Fix data race in executor

### PR DESCRIPTION
**Pull request by ahsanbarkati**
The operations on `inDeg` field of `mutation` struct should be done atomically because it is being used to check if a mutation in dependent on any other mutation. And the mutations are run concurrently, so this update in `inDeg` should be protected. This change fixes the data race which results in alpha crash due to calling `Done` on watermark which is lesser than `doneUntil` 

Fixes: [DGRAPH-2805](https://dgraph.atlassian.net/browse/DGRAPH-2805)

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7203)

### CODE REVIEW

File: worker/executor.go [hunk 1]
```diff
@@ -185,8 +185,8 @@ func (e *executor) worker(mut *mutation) {
 
 	// Decrease inDeg of dependents. If this mutation unblocks them, queue them.
 	for _, dependent := range mut.dependentMutations {
-		dependent.inDeg -= 1
-		if dependent.inDeg == 0 {
+		atomic.AddInt64(&dependent.inDeg, -1)
+		if atomic.LoadInt64(&dependent.inDeg) == 0 {
```
Comment (jarifibrahim) [inline worker/executor.go:189]:
```suggestion
		
		if atomic.AddInt64(&dependent.inDeg, -1) == 0 {
```

**Review by jarifibrahim — APPROVED** [review]
Does this fix the crash we saw @ahsanbarkati ?

**Comment by NamanJain8** [comment]
Can you please add a brief description in PR description about why this should be atomic?

File: worker/executor.go (hunk 1 above)
Comment (ahsanbarkati) [reply worker/executor.go]: It has been running fine since last night. I think this will fix the crash. I'll let it run for a day or two.

**Pull request merged**